api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -97,6 +98,9 @@ message HeaderMutation {
     string remove = 1
         [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
 
+    // Remove the header if it matches.
+    type.matcher.v3.StringMatcher remove_match = 3;
```

Comment:
I think this is a little ambiguous. What you mean by the `it matches`? The key or the value?

Although I think this is a reasonable requirement. I will still prefer the `match and action` type. I mean we can create a matcher and composite the matcher and a list of mutation rules. Just for example:

```
message MatchHeaderMuation {
   repeated config.route.v3.HeaderMatcher headers = 1;
   repeated config.common.mutation_rules.v3.HeaderMutation mutations = 2;
}
```

Reply:
Not sure I follow. The comment is ambiguous yes; I mean to remove headers using a regex/prefix etc whereas HeaderMutation only supports removing a header by the key itself. Is the thought that we should make HeaderMutation more generic? As in, define two new types HeaderMutationMatch and HeaderMutationAction (enum) and nest a list of HeaderMutationMatch with and semantics and a list of HeaderMutationActions?

Reply:
Drive by comment: what is the use case here? For something like ext_proc, presumably the server can just send back individual header mutations and perform the match on its side? Ideally we keep this simple all things being equal.

Reply:
Yes @htuch sorry this might be a case of the XY problem on my part. We ultimately would like to perform something similar to external authorization, but we prefer the auth logic be statically compiled in the Envoy binary rather than in another dependency. It would be an easy sell if there were some field in ext-auth/ext-proc that allowed invoking a extension rather than a grpc service. I was prototyping using an ext-auth grpc service to actually see if it was problematic (my assumption is not but it is a harder sell), though ran out of bandwidth to finish evaluation.
Ultimately we'd like to sanitize a list of headers that is easy to do by matching by a prefix.

Reply:
I'm wondering in this case whether some combination of RBAC + RouteConfiguration would work, or if this is really specialied, maybe just a custom Lua/Wasm/C++ filter.

Reply:
Yes, we already use a C++ filter. I opened the issue just in case it would be helpful for others, so should the issue/pr be closed, it wouldn't really affect us.